language/stdlib/src/transaction_scripts.rs

```diff
@@ -1,53 +1,76 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{compile_script, MOVE_EXTENSION};
+use crate::{
+    compile_script, use_staged, MOVE_EXTENSION, STAGED_EXTENSION, STAGED_OUTPUT_PATH,
+    TRANSACTION_SCRIPTS,
+};
 use once_cell::sync::Lazy;
 use std::path::PathBuf;
 
 // Encodes the absolute path for the transaction scripts directory.
 static TXN_SCRIPTS_DIR: Lazy<PathBuf> = Lazy::new(|| {
     let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
-    path.push(PathBuf::from("transaction_scripts"));
+    path.push(PathBuf::from(TRANSACTION_SCRIPTS));
     path
 });
 
-// Prepends the directory for the transaction scripts and adds the move file suffix to the given
-// source location.
-fn script_source(file_path: &str) -> String {
+// Encodes the absolute path for the staged transaction scripts directory.
+static STAGED_TXN_SCRIPTS_DIR: Lazy<PathBuf> = Lazy::new(|| {
+    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    path.push(STAGED_OUTPUT_PATH);
+    path.push(TRANSACTION_SCRIPTS);
+    path
+});
+
+fn script_source(script_name: &str) -> Vec<u8> {
```

Comment:
The name of this function seems a little misleading, since its result is the compiled binary for the script, not the source.

Reply:
Good point. I'll update the name. I propose `compiled_script_source` unless you have any objections.

Reply:
The "source" part of the name is what seems wrong to me.